packages/type-utils/src/getConstrainedTypeAtLocation.ts

```diff
@@ -4,8 +4,18 @@ import type {
 } from '@typescript-eslint/typescript-estree';
 import type * as ts from 'typescript';
 
+import * as tsApiUtils from 'ts-api-utils';
+
 /**
  * Resolves the given node's type. Will resolve to the type's generic constraint, if it has one.
+ *
+ * If the type is generic and does _not_ have a constraint, the type will be
+ * returned as-is, rather than returning an `unknown` type. This can be checked
+ * for by checking for the type flag ts.TypeFlags.TypeParameter.
+ *
+ * @see https://github.com/typescript-eslint/typescript-eslint/issues/10438
+ *
+ * @deprecated Use getConstraintTypeInfoAtLocation instead.
```

Comment:
[Docs] Fancy linkage:

```suggestion
 * @deprecated Use {@link getConstraintTypeInfoAtLocation} instead.
```